src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -660,8 +666,131 @@ private SpanQuery getSpanQuery(
+            getTermRangeQuery(protoSpanMultiTermQuery.getTermRangeQuery());
+        return new SpanMultiTermQueryWrapper<>(termRangeQuery);
+      default:
+        throw new IllegalArgumentException(
+            "Unsupported Span Multi Query Term Wrapper: " + protoSpanMultiTermQuery);
+    }
+  }
+
+  private static FuzzyQuery getFuzzyQuery(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery) {
+    com.yelp.nrtsearch.server.grpc.FuzzyQuery protoFuzzyQuery =
+        protoSpanMultiTermQuery.getFuzzyQuery();
+    Term term = new Term(protoFuzzyQuery.getField(), protoFuzzyQuery.getText());
+
+    int maxEdits =
+        protoFuzzyQuery.getMaxEdits() == FuzzyQuery.defaultMaxEdits
+            ? FuzzyQuery.defaultMaxEdits
+            : protoFuzzyQuery.getMaxEdits();
+
+    int prefixLength =
+        protoFuzzyQuery.getPrefixLength() == FuzzyQuery.defaultPrefixLength
+            ? FuzzyQuery.defaultPrefixLength
+            : protoFuzzyQuery.getPrefixLength();
+
+    int maxExpansions =
+        protoFuzzyQuery.getMaxExpansions() == 0
+            ? FuzzyQuery.defaultMaxExpansions
+            : protoFuzzyQuery.getMaxExpansions();
+
+    // Set the default transpositions to true, if it is not provided.
+    boolean transpositions =
+        protoFuzzyQuery.hasTranspositions()
+            ? protoFuzzyQuery.getTranspositions()
+            : FuzzyQuery.defaultTranspositions;
+
+    FuzzyQuery fuzzyQuery =
+        new FuzzyQuery(term, maxEdits, prefixLength, maxExpansions, transpositions);
+    fuzzyQuery.setRewriteMethod(
+        getRewriteMethod(protoFuzzyQuery.getRewrite(), protoFuzzyQuery.getRewriteTopTermsSize()));
+    return fuzzyQuery;
+  }
+
+  private static RegexpQuery getRegexpQuery(
+      com.yelp.nrtsearch.server.grpc.SpanMultiTermQuery protoSpanMultiTermQuery) {
+
+    com.yelp.nrtsearch.server.grpc.RegexpQuery protoRegexpQuery =
+        protoSpanMultiTermQuery.getRegexpQuery();
+
+    Term term = new Term(protoRegexpQuery.getField(), protoRegexpQuery.getText());
+
+    int maxDeterminesizedStates = protoRegexpQuery.getMaxDeterminizedStates();
+
+    int flags =
+        switch (protoRegexpQuery.getFlag()) {
+          case REGEXP_ALL -> RegExp.ALL;
+          case REGEXP_ANYSTRING -> RegExp.ANYSTRING;
+          case REGEXP_AUTOMATON -> RegExp.AUTOMATON;
+          case REGEXP_COMPLEMENT -> RegExp.COMPLEMENT;
+          case REGEXP_EMPTY -> RegExp.EMPTY;
+          case REGEXP_INTERSECTION -> RegExp.INTERSECTION;
+          case REGEXP_INTERVAL -> RegExp.INTERVAL;
+          case REGEXP_NONE -> RegExp.NONE;
+          default -> RegExp.ALL;
+        };
+
+    if (maxDeterminesizedStates != 0) {
+      return new RegexpQuery(term, flags, maxDeterminesizedStates);
+    }
+    RegexpQuery regexpQuery = new RegexpQuery(term, flags);
+    regexpQuery.setRewriteMethod(
+        getRewriteMethod(protoRegexpQuery.getRewrite(), protoRegexpQuery.getRewriteTopTermsSize()));
+    return regexpQuery;
```

Comment:
This will skip setting the rewrite method if `maxDeterminesizedStates` is specified.